### PR TITLE
Implement WaitForHttp and WaitForHostPort strategies

### PR DESCRIPTION
**Pull request by rw4lll**
This pull request introduces several significant changes across multiple files to enhance the HTTP and TCP port waiting strategies, improve the handling of HTTP methods, and update the integration tests accordingly.

### Enhancements to HTTP and TCP Port Waiting Strategies:
* [`src/Wait/WaitForHostPort.php`](diffhunk://#diff-dbe2fffb935c7e4c0fd731be1747e9b2bee28090d76d71ba5d932780b982b0f4R1-R51): Added a new `WaitForHostPort` class to wait for a specific host port to be open. It includes methods for checking port status and handling timeouts.
* [`src/Wait/WaitForHttp.php`](diffhunk://#diff-cf8f57e5a397b09a9af66563382a627a297fd108fc9e0b140f74f16ec649772aL7-R147): Refactored the `WaitForHttp` class to extend `BaseWaitStrategy`, added support for various HTTP methods, and included new configuration options like headers, read timeout, and insecure connections.
* [`src/Wait/WaitForTcpPortOpen.php`](diffhunk://#diff-a5065c0df7974a659124d7d28793679dede23691e22594aca440714bd2f3055dL7-R24): Deprecated the `WaitForTcpPortOpen` class in favor of `WaitForHostPort` for better maintainability and consistency.

### Improvements to Handling HTTP Methods:
* [`src/Container/HttpMethod.php`](diffhunk://#diff-c5389ea575ac148c6319c7e7cc0b8059be5979b5b4354ccb88ac18a3b85457d9R1-R28): Introduced an `HttpMethod` enum to standardize HTTP method handling and added a `fromString` method for converting strings to enum values.

### Updates to Integration Tests:
* [`tests/Integration/OldTests/WaitStrategyTest.php`](diffhunk://#diff-e597a28b9c23171fa730af0ceba0eff8bfc992a08f1682aded90266cae08fd3bL24-L29): Updated the integration tests to use the new waiting strategies and removed incomplete test markers. Ensured the tests cover various scenarios for HTTP and TCP port checks. [[1]](diffhunk://#diff-e597a28b9c23171fa730af0ceba0eff8bfc992a08f1682aded90266cae08fd3bL24-L29) [[2]](diffhunk://#diff-e597a28b9c23171fa730af0ceba0eff8bfc992a08f1682aded90266cae08fd3bL89-R111)

### Additional Enhancements:
* [`src/Wait/BaseWaitStrategy.php`](diffhunk://#diff-4c85334f450b1f3596690866faca7a9612eb7f82eb63defcc76241e1db716cdcR16-R27): Added `withTimeout` and `withPollInterval` methods to the `BaseWaitStrategy` class to allow configuration of timeout and polling intervals.

### CODE REVIEW

File: src/Wait/WaitForHttp.php [hunk 1]
```diff
@@ -4,83 +4,146 @@
...
+     * Should be removed in next major version
+     */
     public static function make(int $port): self
     {
-        return new WaitForHttp($port);
+        return new self($port);
     }
 
-    /**
-     * @param WaitForHttp::METHOD_* $method
```
Comment (shyim) [inline src/Wait/WaitForHttp.php:37]: we can keep the php doc right, so phpstan understands what kind of string this is? 🤔

File: src/Container/HttpMethod.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        return match (strtoupper($method)) {
+            'GET' => self::GET,
+            'POST' => self::POST,
+            'PUT' => self::PUT,
+            'DELETE' => self::DELETE,
+            'HEAD' => self::HEAD,
+            'OPTIONS' => self::OPTIONS,
+            default => throw new \InvalidArgumentException("Invalid HTTP method: $method"),
+        };
```
Comment (jdreesen) [inline src/Container/HttpMethod.php:26]:
```suggestion
        return self::from(strtoupper($method));
```

Or if you want to keep the custom exception:
```suggestion
        return self::tryFrom(strtoupper($method)) ?? throw new \InvalidArgumentException("Invalid HTTP method: $method");
```

Comment (rw4lll) [reply src/Container/HttpMethod.php:26]: Yeah, this would be better 😄

File: src/Wait/WaitForHttp.php (hunk 1 above)
Comment (rw4lll) [reply src/Wait/WaitForHttp.php:37]: Yes, I've restored the doc-block. Actually I think that would be better to keep only enum for next major version (currently we have this string type as a workaround for compatibility), but this is the question to discuss in the future.

**Review by shyim — APPROVED** [review]

**Comment by shyim** [comment]
thank you!

**Pull request merged**